url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/agentes/jade/Centralizador/AgenteCentralizador.java

```diff
@@ -0,0 +1,415 @@
+											atuadorIsBusy[Integer.parseInt(indexCurrentAgent)].setApplyParacetamol(true);
+										}
+									}
+
+								}
+							}
+						} catch (Exception e) {
+							System.out.println(getLocalName() + ". Erro ao ler resposta do monitor: " + e.getMessage());
+						}
+					} else if (atuador.contains(sender)){
+						System.out.println(getLocalName() + ": Msg recebida do " + msg.getSender().getLocalName() + " -->" + msg.getContent());
+						updateOutputFile(getLocalName() + ": Msg recebida do " + msg.getSender().getLocalName() + " -->" + msg.getContent(), file);
+					}
+						
+					else {// ERRO
+						System.out.println(getLocalName() + ": "
+								+ sender.getLocalName() + ", "
+								+ msg.getContent());
+					}
+					// System.out.println("**********************");
+					// System.out.println();
+				}
+
+			}
+		});
+	}
+
+	private AID CreatNewAgent(AgentContainer container, String agentName, String agentAdrress) {
+		AID minhaID = null;
+		try {
+			System.out.println(getLocalName() + ": Agente " + agentName + " criado com sucesso");
+			container.createNewAgent(agentName, agentAdrress, null).start();
+		} catch (StaleProxyException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return minhaID;
+	}
+
+	private void sendMessageToAgent(String mensagem, String agentName) {
+		ACLMessage message = new ACLMessage(ACLMessage.REQUEST);
+		AID receiver = new AID(agentName, AID.ISLOCALNAME);
+		message.addReceiver(receiver);
+		message.setContent(mensagem);
+		System.out.println(getLocalName() + ": " + receiver.getLocalName() + ", " + message.getContent());
+		updateOutputFile(getLocalName() + ": " + receiver.getLocalName() + ", " + message.getContent(), file);
+		//System.out.println(getLocalName() + "para " + agentName + " : " + mensagem);
+		send(message);
+	}
+	
+	private String getReferenceValue(String tipo){
+		StringBuilder result = new StringBuilder();
+		result.append(". Valor de referencia: ");
+		if (tipo == "temperatura")
+			result.append("35<T<" + String.valueOf(INICIO_FEBRE));
+		else if (tipo == "hemoglobina")
+			result.append(String.valueOf(MIN_HEMO_BOA) + "<H<" + String.valueOf(MAX_HEMO_BOA));
+		else if (tipo == "bilirrubina")
+			result.append(String.valueOf(MIN_BILI_BOA) + "<B<" + String.valueOf(MAX_BILI_BOA));
+		else if (tipo == "pressao") {
+			result.append("Sist -> " + String.valueOf(MIN_PRESSAO_SIST_BOA) + "<SIST<" + String.valueOf(MAX_PRESSAO_SIST_BOA));
+			result.append(". Diast -> " + String.valueOf(MIN_PRESSAO_DIAS_BOA) + "<DIAST<" + String.valueOf(MAX_PRESSAO_DIAS_BOA));
+			}
+		else System.out.println("Erro. Tipo desconhecido: " + tipo);
+		
+		return result.toString();
+	}
+	
+
+	private void updateOutputFile(String data, File file) {
+		// if file doesnt exists, then create it
+		// if(!file.exists()){
+		// file.createNewFile();
+		// }
+
+		// true = append file
+		FileWriter fileWritter;
+		try {
+			fileWritter = new FileWriter(file.getName(), true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fileWritter*. The resource is referenced at lines 401, 403, 404. The resource is closed at line 405. There are other execution paths that do not contain closure statements, for example, when BufferedWriter.newLine throws an exception. Ensure *fileWritter* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).